### PR TITLE
Move SegNext models from experimental to custom models

### PR DESCRIPTION
**Pull request by kprokofi**
### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Comment by kprokofi** [comment]
image
e2e passed for SegNext-t

**Review by goodsong81 — CHANGES_REQUESTED** [review]
Thank you for the great work, @kprokofi.
BTW, Mark decided not to update the default det/seg models in Geti1.8 timeline for safety. The classification model, DEIT, only will be added as custom model. So, let's target this PR (and requirement) to OTX1.5. We can merge this PR to develop after creating releases/1.4.0.
c.c. @sovrasov

**Comment by sovrasov** [comment]
@kprokofi please update the PR. Now we have `model_category` in template to map models to Geti tasks, so experimental_ prefix doesn't make sense anymore.

File: tests/e2e/cli/semantic_segmentation/test_segmentation.py [hunk 1]
```diff
@@ -317,6 +317,8 @@ class TestToolsMPASelfSLSegmentation:
     @pytest.mark.parametrize("template", templates, ids=templates_ids)
     def test_otx_train(self, template, tmp_dir_path):
         tmp_dir_path_1 = tmp_dir_path / "segmentation/test_selfsl"
+        if not os.path.exists(os.path.join(template.model_template_path[:-14], "selfsl")):
```
Comment (sovrasov) [inline tests/e2e/cli/semantic_segmentation/test_segmentation.py]: We agreed to use Pathlib in OTX. Also, `path.parent` may solve the slitting issue without indexing lists

File: tests/e2e/cli/semantic_segmentation/test_segmentation.py [hunk 2]
```diff
@@ -341,11 +340,12 @@ def test_otx_eval(self, template, tmp_dir_path):
     @pytest.mark.skipif(MULTI_GPU_UNAVAILABLE, reason="The number of gpu is insufficient")
     @pytest.mark.parametrize("template", templates, ids=templates_ids)
     def test_otx_multi_gpu_train_selfsl(self, template, tmp_dir_path):
-        if not os.path.exists(os.path.join(template.model_template_path[:-14], "selfsl")):
-            pytest.skip("SelfSL isn't available for this template")
+        # if Path(template.model_template_path, )
```
Comment (sovrasov) [inline tests/e2e/cli/semantic_segmentation/test_segmentation.py]: Please del commented code

**Review by sungmanc — APPROVED** [review]
LGTM for me, have two simple questions.

1. Is there any reason to change the `os` --> `path` to represent the file path and directory?
2. Is `early_stop_patience` value set by experiments? It seems that it's quite large if we use the large dataset

**Comment by kprokofi** [comment]
> LGTM for me, have two simple questions.
> 
> 1. Is there any reason to change the `os` --> `path` to represent the file path and directory?
> 2. Is `early_stop_patience` value set by experiments? It seems that it's quite large if we use the large dataset

1. We agreed to use Pathlib instead of os.path, so I started with segmentation and changed os to path. Thre is no other reasons.
2.  After experiments on small datasets (10-40 images) it turned out that SegNext tends to show around 0 metric during first epochs and training stops because of early stopping. However, if we train model with more epochs - the results got much better (as expected). So, I increased early stopping lower boundary. 
Since our main target is Geti, it seems reasonable. On large datasets first 15 epochs without early stopping looks not too much, I think. If it cause some concerns and training time relate issues, I can think about some heuristic based on dataset length in the future

**Review by sovrasov — APPROVED** [review]

**Comment by sungmanc** [comment]
> > LGTM for me, have two simple questions.
> > 
> > 1. Is there any reason to change the `os` --> `path` to represent the file path and directory?
> > 2. Is `early_stop_patience` value set by experiments? It seems that it's quite large if we use the large dataset
> 
> 1. We agreed to use Pathlib instead of os.path, so I started with segmentation and changed os to path. Thre is no other reasons.
> 2. After experiments on small datasets (10-40 images) it turned out that SegNext tends to show around 0 metric during first epochs and training stops because of early stopping. However, if we train model with more epochs - the results got much better (as expected). So, I increased early stopping lower boundary.
>    Since our main target is Geti, it seems reasonable. On large datasets first 15 epochs without early stopping looks not too much, I think. If it cause some concerns and training time relate issues, I can think about some heuristic based on dataset length in the future

Thanks for your replying ! BTW, what I concerned about is `early_stop_patience`, not `early_stop_start`. However, it's not important for now.

**Comment by kprokofi** [comment]
@goodsong81 Could you approve, please?

**Review by goodsong81 — APPROVED** [review]
Oh, sorry for late review. I thought I've done already.
LGTM! :)

**Pull request merged**